ci: Error: RuboCop found unsupported Ruby version 2.4

## My Environment

| Software         | Version(s)  |
| ---------------- | ----------- |
| Operating System | Artix Linux |
| `jekyll`         | Latest      |
| `github-pages`   | Latest      |

---

## Expected Behaviour

When submitting a PR, style-check works.

From #8646 / <https://github.com/jekyll/jekyll/runs/2386310191?check_suite_focus=true>:

<details>
<summary>Log of the last ci job that worked</summary>

```plaintext
        +style-check | RUBY=2.5
        +style-check | --> RUN script/fmt
        +style-check | RuboCop 1.12.1
        +style-check | The following cops were added to RuboCop, but are not configured. Please set Enabled to either `true` or `false` in your `.rubocop.yml` file.

        +style-check | Please also note that can also opt-in to new cops by default by adding this to your config:
        +style-check |   AllCops:
        +style-check |     NewCops: enable
        +style-check | Minitest/AssertWithExpectedArgument: # (new in 0.11)
        +style-check |   Enabled: true
        +style-check | Performance/RedundantEqualityComparisonBlock: # (new in 1.10)
        +style-check |   Enabled: true
        +style-check | Performance/RedundantSplitRegexpArgument: # (new in 1.10)
        +style-check |   Enabled: true
        +style-check | For more information: https://docs.rubocop.org/rubocop/versioning.html
        +style-check | Inspecting 134 files
        +style-check | ......................................................................................................................................

        +style-check | 134 files inspected, no offenses detected
              output | --> exporting outputs
=================================== SUCCESS ====================================
```

</details>

## Current Behavior

style-check fails because rubocop dies.

From #8646 / <https://github.com/jekyll/jekyll/runs/2394239794?check_suite_focus=true>

<details>
<summary>Log of the first ci job that failed</summary>

```plaintext
        +style-check | RUBY=2.5
        +style-check | --> RUN script/fmt
        +style-check | RuboCop 1.13.0
        +style-check | Error: RuboCop found unsupported Ruby version 2.4 in `TargetRubyVersion` parameter (in .rubocop.yml). 2.4-compatible analysis was dropped after version 1.12.
        +style-check | Supported versions: 2.5, 2.6, 2.7, 3.0

        +style-check | Try running `script/fmt -a` to automatically fix errors
        +style-check | Command /bin/sh -c script/fmt failed with exit code 2
        +style-check | 
        +style-check | ERROR: Command exited with non-zero code: RUN script/fmt
Repeating the output of the command that caused the failure
================================ FAILURE [main] ================================
        +style-check *failed* | ERROR: Command exited with non-zero code: RUN script/fmt
        +style-check *failed* | RUBY=2.5
Error: bkClient.Build: failed to solve: executor failed running [/bin/sh -c  /usr/bin/earth_debugger /bin/sh -c 'script/fmt']: exit code: 2
        +style-check *failed* | --> RUN script/fmt
        +style-check *failed* | RuboCop 1.13.0
        +style-check *failed* | Error: RuboCop found unsupported Ruby version 2.4 in `TargetRubyVersion` parameter (in .rubocop.yml). 2.4-compatible analysis was dropped after version 1.12.
        +style-check *failed* | Supported versions: 2.5, 2.6, 2.7, 3.0

        +style-check *failed* | Try running `script/fmt -a` to automatically fix errors
        +style-check *failed* | Command /bin/sh -c script/fmt failed with exit code 2
        +style-check *failed* | 
Error: Process completed with exit code 1.
```

</details>

Key parts:

> RuboCop 1.13.0
>
> Error: RuboCop found unsupported Ruby version 2.4 in `TargetRubyVersion`
> parameter (in .rubocop.yml). 2.4-compatible analysis was dropped after
> version 1.12.

## Code Sample

```console
$ bundle exec rubocop --version
1.13.0
$ bundle exec rubocop
Error: RuboCop found unsupported Ruby version 2.4 in `TargetRubyVersion` parameter (in .rubocop.yml). 2.4-compatible analysis was dropped after version 1.12.
Supported versions: 2.5, 2.6, 2.7, 3.0
```

Note: Jekyll dropping Ruby 2.4 had already been planned:

* <https://github.com/jekyll/jekyll/issues/8116>
* <https://github.com/jekyll/jekyll/issues/8115>

See also:

* <https://github.com/rubocop/rubocop/releases/tag/v1.13.0>
* <https://github.com/rubocop/rubocop/pull/9648>

--- a/Gemfile
+++ b/Gemfile
@@ -23,7 +23,7 @@ group :test do
   gem "nokogiri", "~> 1.7"
   gem "rspec"
   gem "rspec-mocks"
-  gem "rubocop", "~> 1.10"
+  gem "rubocop", "~> 1.12.0"
   gem "rubocop-minitest"
   gem "rubocop-performance"
   gem "rubocop-rake"

